Apromore-Extras/APMLogModule/src/test/java/org/apromore/apmlog/AttributeArcDurationTest.java

```diff
@@ -252,4 +255,39 @@ public static void testRetain3(APMLog apmLog, APMLogUnitTest parent) throws Unsu
             parent.printString("'Attribute arc duration (3)' test PASS.\n");
         }
     }
+
+    public static void testAvgDur1(APMLog apmLog, APMLogUnitTest parent) throws UnsupportedEncodingException {
+        APMLogFilter apmLogFilter = new APMLogFilter(apmLog);
+        PLog pLog = apmLogFilter.getPLog();
+        pLog.updateStats();
+        DurSubGraph subGraph = pLog.getAttributeGraph().getNextValueDurations("concept:name", "A");
+        DoubleArrayList durList = subGraph.getValDurListMap().get("A");
+        double avg = getAverage(durList);
+        double twoHours = 1000 * 60 * 60 * 2d;
+        if (avg != twoHours) {
+            throw new AssertionError("TEST FAILED. RESULT MISMATCH.\n expected value=" + twoHours +
+                    "; output value=" + avg);
+        } else {
+            parent.printString("'Attribute arc duration (4) - Average duration' test PASS.\n");
+        }
+    }
+
+    public static double getAverage(DoubleArrayList doubleList) {
+        if (doubleList.size() > 0) {
+            double sum = getTotal(doubleList);
+            return doubleList.size() > 1 ? sum / doubleList.size() : sum;
+        }
+        return 0;
+    }
```

Comment:
I think there's a provided DoubleArrayList.average method to do this without an extra method.

Reply:
updated